File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public String solution(String my_string, String overwrite_string, int s) {
        String answer = "";
        StringBuilder sb = new StringBuilder("");
        
        int over_len = overwrite_string.length();
            
        for(int i=0;i<my_string.length();i++){
            
            if(i<s||over_len+s<=i){
                sb.append(my_string.charAt(i));
            }else{
                sb.append(overwrite_string.charAt(i-s));
            }
        }
        
        
        return sb.toString();
    }
}

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.kt
class Solution {
    fun solution(my_string: String, overwrite_string: String, s: Int): String {
        var answer: String = ""
        
        for(i in 0..s-1){
            answer+=my_string[i]
        }
        
        for(c in overwrite_string){
            answer+=c
        }
        
        for(i in overwrite_string.length+s..my_string.length-1){
            answer+=my_string[i]
        }
       	
        
        return answer
    }
}